core/server/worker/src/main/java/alluxio/worker/dora/RocksDBDoraMetaStore.java

```diff
@@ -0,0 +1,187 @@
+import java.util.stream.Collectors;
+
+/**
+ * Dora Meta Store in RocksDB.
+ */
+public class RocksDBDoraMetaStore implements DoraMetaStore {
+  private static final Logger LOG = LoggerFactory.getLogger(RocksDBDoraMetaStore.class);
+  private static final String DORA_META_DB_NAME = "DoraMeta";
+  private static final String DORA_META_FILE_STATUS_COLUMN = "FileStatusCF";
+  private static final String DORA_META_STORE_NAME = "DoraMetaStore";
+
+  // These are fields instead of constants because they depend on the call to RocksDB.loadLibrary().
+  private final WriteOptions mWriteOption;
+  private final ReadOptions mReadOption;
+  private final RocksStore mRocksStore;
+  private final List<RocksObject> mToClose = new ArrayList<>();
+
+  private final AtomicReference<ColumnFamilyHandle> mFileStatusColumn = new AtomicReference<>();
+
+  /**
+   * Creates and initializes a rocks block store.
+   *
+   * @param baseDir the base directory in which to store inode metadata
+   */
+  public RocksDBDoraMetaStore(String baseDir) {
+    RocksDB.loadLibrary();
+    // the rocksDB objects must be initialized after RocksDB.loadLibrary() is called
+    mWriteOption = new WriteOptions();
+    mReadOption  = new ReadOptions();
+    String dbPath = PathUtils.concatPath(baseDir, DORA_META_DB_NAME);
+    String backupPath = PathUtils.concatPath(baseDir, DORA_META_DB_NAME + "-backup");
+
+    List<ColumnFamilyDescriptor> columns = new ArrayList<>();
+    DBOptions opts = new DBOptions();
+    mToClose.add(opts);
+    mToClose.add(mReadOption);
+    mToClose.add(mWriteOption);
+
+    opts.setAllowConcurrentMemtableWrite(false) // not supported for hash mem tables
+            .setCreateMissingColumnFamilies(true)
+            .setCreateIfMissing(true)
+            .setMaxOpenFiles(-1);
+    columns.add(new ColumnFamilyDescriptor(DORA_META_FILE_STATUS_COLUMN.getBytes(),
+            new ColumnFamilyOptions()
+                    .setMemTableConfig(new HashLinkedListMemTableConfig())
+                    .setCompressionType(CompressionType.NO_COMPRESSION)));
+    mToClose.addAll(columns.stream().map(
+            ColumnFamilyDescriptor::getOptions).collect(Collectors.toList()));
+
+    mRocksStore = new RocksStore(DORA_META_STORE_NAME, dbPath, backupPath, opts, columns,
+            Arrays.asList(mFileStatusColumn), false);
+  }
+
+  /**
+   * Queries the metadata for a file.
+   *
+   * @param path the full path of this file
+   * @return the metadata if found, Optional.empty if not found
+   */
+  @Override
+  public Optional<DoraMeta.FileStatus> getDoraMeta(String path) {
+    byte[] status;
+    try {
+      status = db().get(mFileStatusColumn.get(), path.getBytes());
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+    if (status == null) {
+      return Optional.empty();
+    }
+    try {
+      return Optional.of(DoraMeta.FileStatus.parseFrom(status));
+    } catch (Exception e) {
+      throw new RuntimeException(e);
```

Comment:
will end-user get this RuntimeException or where is the final place deal with it?

Reply:
This is a good point.
A better error handling is needed here.
If some exception happens here, we should return Optional.empty() and delete this record.

I will fix it.

Reply:
I removed the `throw Exception`, and returns an Optional.empty().
In this way, this is the final place to consume the exception.
But now, the caller will not know what kind of error happened or there is no such record.

Please give more comments about this handling.